src/Microsoft.DotNet.Interactive/KernelBase.cs

```diff
@@ -179,6 +183,60 @@ private async Task SetKernel(IKernelCommand command, KernelInvocationContext con
             }
         }
 
+        private async Task HandleLinePositionLanguageServiceRequests(
+            LanguageServiceCommandBase languageServiceCommand,
+            KernelInvocationContext context,
+            KernelPipelineContinuation continueOnCurrentPipeline)
+        {
+            var tree = SubmissionParser.Parse(languageServiceCommand.Code, languageServiceCommand.TargetKernelName);
+            var nodes = tree.GetRoot().ChildNodes.ToArray();
+            if (nodes.Length == 1)
```

Comment:
Rather than inferring this (as it might not be future-proof as the parser evolves), we might generalize the approach taken by `SubmitCode.SuppressSplit`.